src/VirtoCommerce.CatalogModule.Data/Repositories/CatalogRepositoryImpl.cs

```diff
@@ -530,14 +529,50 @@ public virtual async Task<GenericSearchResult<AssociationEntity>> SearchAssociat
             return result;
         }
 
+        public virtual async Task<ICollection<CategoryEntity>> SearchCategoriesHierarcyAsync(string categoryId)
+        {
+            var commandTemplate = @"
+                WITH CategoryParents AS   
+                (  
+                    SELECT * 
+                    FROM Category   
+                    WHERE Id = @categoryId
+                    UNION ALL  
+                    SELECT c.*
+                    FROM Category c, CategoryParents cp
+	                where c.Id = cp.ParentCategoryId 
+                )  
+                SELECT *
+                FROM CategoryParents";
+
+            var categoryIdParam = new SqlParameter("@categoryId", categoryId);
```

Comment:
Can we use LoadByIds to load found categories, instead of this code duplication?